include/deal.II/base/array_view.h

```diff
@@ -305,13 +332,32 @@ namespace internal
 
 
 
+template <typename ElementType, typename MemorySpaceType>
+inline ArrayView<ElementType, MemorySpaceType>::ArrayView()
+  : starting_element(nullptr)
+  , n_elements(0)
+{}
+
+
+
 template <typename ElementType, typename MemorySpaceType>
 inline ArrayView<ElementType, MemorySpaceType>::ArrayView(
   value_type *      starting_element,
   const std::size_t n_elements)
-  : starting_element(starting_element)
-  , n_elements(n_elements)
 {
+  reinit(starting_element, n_elements);
+}
+
+
+
+template <typename ElementType, typename MemorySpaceType>
+inline void
+ArrayView<ElementType, MemorySpaceType>::reinit(value_type *starting_element_,
+                                                const std::size_t n_elements_)
+{
+  starting_element = starting_element_;
+  n_elements       = n_elements_;
```

Comment:
If you don't like the code duplication with what I wanted to leave in place above, you can simply write this as follows:
```
  *this = ArrayView(starting_element, n_element);
```
Then you also don't need the underscore on the name. You can also put the assertion you have here into the constructor.